libreoffice-core/vcl/source/outdev/font.cxx

```diff
@@ -384,6 +384,45 @@ void OutputDevice::BeginFontSubstitution()
     pSVData->maGDIData.mbFontSubChanged = false;
 }
 
+void OutputDevice::AddCustomMacroFonts()
+{
+    OutputDevice* pDevice = Application::GetDefaultDevice();
+    pDevice->ImplAddCustomMacroFonts();
+}
+
+void OutputDevice::ImplAddCustomMacroFonts()
+{
+    std::list<std::string> fontNames = {
+        "Carlito",
+        "Carlito-Bold",
+        "Carlito-Italic",
+        "Carlito-BoldItalic",
+        "Caladea",
+        "Caladea-Bold",
+        "Caladea-Italic",
+        "Caladea-BoldItalic"
+    };
+
+    // get proper build link here
+    std::string basePath = "file://tk";
+    std::string extension = ".ttf";
+    std::string affix = "_MACRO";
+
+    bool allLoaded = true;
+    for (std::string name : fontNames) {
+        OUString filePath = OUString::createFromAscii(
+            (basePath + name + extension).data()
+        );
+        OUString customName = OUString::createFromAscii(
+            (name + affix).data()
+        );
+        OutputDevice::AddTempDevFont(
```

Comment:
I am going to benchmark this a bit- it is only 8 fonts but I feel like it does add a small tax. Probably can yank this code into a batched op.